pkg/gadgets/dns/gadget.go

```diff
@@ -39,14 +40,23 @@ type Trace struct {
 	started bool
 
 	tracer *dnstracer.Tracer
+
+	netnsHost uint64
 }
 
 type TraceFactory struct {
 	gadgets.BaseFactory
+
+	netnsHost uint64
 }
 
 func NewFactory() gadgets.TraceFactory {
-	return &TraceFactory{}
+	netnsHost, _ := containerutils.GetNetNs(os.Getpid())
+	f := &TraceFactory{
+		netnsHost: netnsHost,
+	}
+	f.DeleteTrace = f.deleteTrace
```

Comment:
> Why are you adding this one outside instead of directly in the struct?

It was because it is a method, so I needed a reference on `f`. But I will change it to a regular function.